### PR TITLE
add support for sitemap generation

### PR DESCRIPTION
**Pull request by tglman**
This implement the sitemap support with the use of sitemap crate, as wrote in this issue: https://github.com/cobalt-org/cobalt.rs/issues/452

### CODE REVIEW

File: src/cobalt.rs [hunk 1]
```diff
@@ -450,3 +464,37 @@ fn create_jsonfeed(
...
 }
+fn create_sitemap(
+    path: &str,
+    dest: &path::Path,
+    collection: &Collection,
+    documents: &[Document],
+    collection_pages: &Collection,
+    documents_pages: &[Document],
+) -> Result<()> {
+    let sitemap_path = dest.join(path);
```
Comment (epage) [inline src/cobalt.rs]: From an API perspective (even if internal) I think I'd prefer this to be in the caller

File: src/cobalt_model/config.rs [hunk 2]
```diff
@@ -67,7 +67,7 @@ impl PageConfig {
             order: collection::SortOrder::None,
             rss: None,
             jsonfeed: None,
-            base_url: None,
+            base_url: site.base_url.clone(),
```
Comment (epage) [inline src/cobalt_model/config.rs]: I'm confused as to why the code didn't do this in the first place.  I'll need to dig into this

File: src/document.rs [hunk 3]
```diff
@@ -264,6 +264,21 @@ impl Document {
...
+    pub fn to_sitemap<T: std::io::Write>(
+        &self,
+        root_url: &str,
+        writer: &mut sitemap::writer::UrlSetWriter<T>,
+    ) -> Result<()> {
+        let link = format!("{}/{}", root_url, &self.url_path);
+        let mut url = sitemap::structs::UrlEntry::builder();
+        url = url.loc(link);
+        if let Some(date) = self.front.published_date {
+            url = url.lastmod(*date);
```
Comment (epage) [inline src/document.rs]: If/when we add a modified_date, we'll need to remember to update this

I think the only issue related to the topic is https://github.com/cobalt-org/cobalt.rs/issues/197

File: src/cobalt_model/config.rs (hunk 2 above)
Comment (epage) [reply src/cobalt_model/config.rs]: Ok, this was previously needed only for RSS, so we just avoided sending along dead data.

**Comment by tglman** [comment]
Hi, 
now should be all fine.

Bye

**Pull request merged**